server/src/main/java/org/elasticsearch/index/codec/bloomfilter/ES85BloomFilterPostingsFormat.java

```diff
@@ -572,4 +609,337 @@ private static int murmurhash3_x86_32(byte[] data, int offset, int len, int seed
 
         return h1;
     }
+
+    // A self-contained implementation of CityHash that is used for the Bloom
+    // filter.
+    /*
+     * Copyright (C) 2012 tamtam180
+     *
+     * Licensed under the Apache License, Version 2.0 (the "License");
+     * you may not use this file except in compliance with the License.
+     * You may obtain a copy of the License at
+     *
+     * http://www.apache.org/licenses/LICENSE-2.0
+     *
+     * Unless required by applicable law or agreed to in writing, software
+     * distributed under the License is distributed on an "AS IS" BASIS,
+     * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     * See the License for the specific language governing permissions and
+     * limitations under the License.
+     */
+
+    /**
+     * @author tamtam180 - kirscheless at gmail.com
+     * see http://google-opensource.blogspot.jp/2011/04/introducing-cityhash.html
+     * see http://code.google.com/p/cityhash/
+     */
+    @SuppressWarnings("fallthrough")
+    public static class CityHash {
+
+        private static final long k0 = 0xc3a5c85c97cb3127L;
+        private static final long k1 = 0xb492b66fbe98f273L;
+        private static final long k2 = 0x9ae16a3b2f90404fL;
+        private static final long k3 = 0xc949d7c7509e6557L;
+
+        private static long toLongLE(byte[] b, int i) {
+            return (((long) b[i + 7] << 56) + ((long) (b[i + 6] & 255) << 48) + ((long) (b[i + 5] & 255) << 40) + ((long) (b[i + 4] & 255)
+                << 32) + ((long) (b[i + 3] & 255) << 24) + ((b[i + 2] & 255) << 16) + ((b[i + 1] & 255) << 8) + ((b[i + 0] & 255) << 0));
+        }
+
+        private static int toIntLE(byte[] b, int i) {
+            return (((b[i + 3] & 255) << 24) + ((b[i + 2] & 255) << 16) + ((b[i + 1] & 255) << 8) + ((b[i + 0] & 255) << 0));
+        }
+
+        private static long fetch64(byte[] s, int pos) {
+            return toLongLE(s, pos);
+        }
+
+        private static int fetch32(byte[] s, int pos) {
+            return toIntLE(s, pos);
+        }
+
+        private static long rotate(long val, int shift) {
+            return shift == 0 ? val : (val >>> shift) | (val << (64 - shift));
+        }
+
+        private static long rotateByAtLeast1(long val, int shift) {
+            return (val >>> shift) | (val << (64 - shift));
```

Comment:
You can use `Long.rotateRight` for this. (FYI shifts on longs are mod 64 in Java so `<< 64 - shift` is the same as `<< -shift`.)

Reply:
Done.